src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -249,7 +249,7 @@ class Ledger(vm: VM) extends Logger {
   private[ledger] def updateSenderAccountBeforeExecution(stx: SignedTransaction, worldStateProxy: InMemoryWorldStateProxy): InMemoryWorldStateProxy = {
     val senderAddress = stx.senderAddress
     val account = worldStateProxy.getGuaranteedAccount(senderAddress)
-    worldStateProxy.saveAccount(senderAddress, account.increaseBalance(-calculateUpfrontCost(stx.tx)).increaseNonce)
+    worldStateProxy.saveAccount(senderAddress, account.increaseBalance(-calculateUpfrontGas(stx.tx)).increaseNonce)
```

Comment:
Is calculate `calculateUpfrontCost` still used?

I still think that `ProgramContext` should not be tasked with such updates (accounts balance). But we can leave, I don't want to focus prematurely on refactoring either.